packages/runtime/container-runtime/src/summarizerHeuristics.ts

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Timer } from "@fluidframework/common-utils";
+import { ISummaryConfiguration } from "@fluidframework/protocol-definitions";
+import { ISummarizerHeuristics, ISummaryAttempt } from "./summarizerTypes";
+import { SummarizeReason } from "./summaryGenerator";
+
+const minOpsForLastSummary = 50;
+
+class BaseSummarizerHeuristics {
```

Comment:
I think this should also implement `ISummarizerHeuristics`. The methods that don't have a base implementation can be marked as abstract.

Reply:
Sorry! Please look again... I made some changes to how this works.